src/DataStructures/Tensor/Expressions/KroneckerDeltaAsExpression.hpp

```diff
@@ -0,0 +1,174 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines expressions that represent the Kronecker delta
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/Tensor/Expressions/TensorExpression.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Symmetry.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace tenex {
+using KroneckerDeltaFrame = Frame::NoFrame;
```

Comment:
I didn't assign a frame because I thought on its own it didn't have framed indices (thought it was basically the identity matrix). If you think it makes sense conceptually for the indices to have a frame and/or to protect against people abusing this to change frames in a product or something, then I'm happy to add it.

Reply:
It most certain does have a frame. It's only the identity on one frame. If you transform to another frame it's no longer the identity